happi/backends/json_db.py

```diff
@@ -138,27 +136,30 @@ def store(self, db: dict[str, ItemMeta]) -> None:
             Dictionary to store in JSON.
         """
         temp_path = self._temp_path()
+        try:
+            with open(temp_path, 'w') as fd:
+                json.dump(db, fd, sort_keys=True, indent=4)
 
-        with open(temp_path, 'w') as fd:
-            json.dump(db, fd, sort_keys=True, indent=4)
+            if os.path.exists(self.path):
+                shutil.copymode(self.path, temp_path)
+            shutil.move(temp_path, self.path)
+        except BaseException as ex:
 
-        if os.path.exists(self.path):
-            shutil.copymode(self.path, temp_path)
-        shutil.move(temp_path, self.path)
+            # log that the db move was not executed properly
+            logger.debug('JSON db move failed: %s', ex, exc_info=ex)
+            # remove temporary file
+            if os.path.exists(temp_path):
+                os.remove(temp_path)
```

Comment:
After this line, we should re-raise the exception so that the caller can handle it, that is:
```python
                os.remove(temp_path)
            raise
```

Reply:
I think this is especially important here because we're catching `BaseException` which might be e.g. a `KeyboardInterrupt`